load_confounds/compcor.py

```diff
@@ -0,0 +1,65 @@
+"""Helper function for _load_compcor."""
+
+
+prefix_compcor = {"full": ["t", "a"], "temp": ["t"], "anat": ["a"]}
+anat_masker = {True: ["combined"], False: ["WM", "CSF"], None: None}
+
+
+def _find_compcor(confounds_json, compcor, n_compcor, acompcor_combined):
+    """Builds list for the number of compcor components."""
+    prefix_set, anat_mask = _check_compcor_method(compcor, acompcor_combined)
+
+    collector = []
+    for prefix in prefix_set:
+        # all possible compcor confounds, mixing different types of mask
+        all_compcor_name = [
+            comp for comp in confounds_json.keys() if f"{prefix}_comp_cor" in comp
+        ]
+        # filter by prefix first and apply acompor mask option if relevant
+        compcor_cols_filt = _prefix_confound_filter(prefix, all_compcor_name)
+        if prefix == "a":
+            compcor_cols_filt = _acompcor_mask(
+                confounds_json, anat_mask, compcor_cols_filt
+            )
+        collector += _select_compcor(compcor_cols_filt, n_compcor)
```

Comment:
@htwangtw that's the change I implemented: extracting the number of selected components per mask, rather than on the aggregate of all confounds. Thinking of it, I would need to check if it works when the masks are not combined too.... so many variations! Please let me know what you think.

Reply:
Yes you are absolutely correct I missed this one! It would be good to have some test cases dedicated to this. Also, in the case of a combined mask, when we request get N compcor, are we expecting the method to return N for WM and N for CSF or N/2 for each?

Reply:
There was indeed a bug when the masks were not combined: it picked the components from the first mask. I decide to interpret `n_compcor` as the number of components per mask (because that's already how it behaved with `"full"` and it wasn't clear what to do for `compcor="full"` and `acompcor_combined=False`). I've added this information in the docstring.  I think everything is covered now.